examples/preprocessing/plot_count_featurizer.py

```diff
@@ -0,0 +1,138 @@
+
+
+def select_discrete(X):
+    """Selects the discrete features."""
+    return X[:, discretized_features]
+
+
+def process_discrete(X):
+    """Processes discrete features to make them categorical."""
+    for feature in discretized_features:
+        X_transform_col = (X[:, feature]).astype(int)
+        col_min = min(np.amin(X_transform_col), 0)
+        X[:, feature] = X_transform_col - col_min
+    return X
+
+
+time_start = time.time()
+pipeline_cf = make_pipeline(
+    FunctionTransformer(func=process_discrete),
+    CountFeaturizer(inclusion=discretized_features),
+    FunctionTransformer(func=lambda X: select_non_discrete(X, count=True)))
+X_count = pipeline_cf.fit_transform(X, y=y)
+cf_time_preprocessing = time.time() - time_start
+
+time_start = time.time()
+pipeline_ohe_nd = make_pipeline(FunctionTransformer(func=select_non_discrete))
+pipeline_ohe_d = make_pipeline(
+    FunctionTransformer(func=select_discrete),
+    FunctionTransformer(func=process_discrete),
+    OneHotEncoder())
+pipeline_ohe = FeatureUnion(
+    [("discrete", pipeline_ohe_d), ("nondiscrete", pipeline_ohe_nd)])
+X_one_hot = pipeline_ohe.fit_transform(X, y=y).todense()
+ohe_time_preprocessing = time.time() - time_start
+
+
+def get_classifier():
+    return RandomForestClassifier(warm_start=True, max_features="log2",
+                                  oob_score=True, random_state=RANDOM_STATE)
+
+
+clf = get_classifier()
+labels = ["CountFeaturizer + RandomForestClassifier",
+          "OneHotEncoder + RandomForestClassifier",
+          "Only RandomForestClassifier"]
+error_rate = OrderedDict((label, []) for label in labels)
+
+min_estimators = (15 * n_datapoints // 500)
+max_estimators = (175 * n_datapoints // 500)
+time_start = time.time()
+
+for i in range(min_estimators, max_estimators + 1):
+    clf.set_params(n_estimators=i)
+    clf.fit(X_count, y)
+    oob_error = 1 - clf.oob_score_
+    error_rate[labels[0]].append((i, oob_error))
+
+print("Time taken on CountFeaturizer: ",
+      (time.time() - time_start + cf_time_preprocessing))
+clf = get_classifier()
+time_start = time.time()
+
+for i in range(min_estimators, max_estimators + 1):
+    clf.set_params(n_estimators=i)
+    clf.fit(X_one_hot, y)
+    oob_error = 1 - clf.oob_score_
+    error_rate[labels[1]].append((i, oob_error))
+
+print("Time taken on OneHotEncoder: ",
+      (time.time() - time_start + ohe_time_preprocessing))
+clf = get_classifier()
+time_start = time.time()
+
+for i in range(min_estimators, max_estimators + 1):
+    clf.set_params(n_estimators=i)
+    clf.fit(X, y)
+    oob_error = 1 - clf.oob_score_
+    error_rate[labels[2]].append((i, oob_error))
+
+print("Time taken on No Encoding: ", (time.time() - time_start))
+
+# Generate the "OOB error rate" vs. "n_estimators" plot.
+for label, clf_err in error_rate.items():
+    xs, ys = zip(*clf_err)
+    plt.plot(xs, ys, label=label)
+
+plt.xlim(min_estimators, max_estimators)
+plt.xlabel("n_estimators")
```

Comment:
(* Personally I'm a big fan of declaring the plotting labels and other stuff (l135, 136, 137)  high up in the code, so
the script is reusable. for this it would be even better to put everything in a function but maybe this is me taking it
too far .. )